Create a general pull_request_template.md (#703)
devicons/devicon: .github/drafts/PULL_REQUEST_TEMPLATE/new_feature.md

---
name: New Feature
about: Add a new feature to the repository.
title: 'New Feature: [NAME]'
labels: 'enhancement'
assignees: ''

---
<!-- If you are adding a new icon, follow the steps listed here, and delete the **New Icon** section above. -->

## Double check these details before you open a PR**

<!-- Tick the checkboxes to ensure you've done everything correctly -->
- [] PR does not match another non-stale PR currently opened
- [] PR name matches the format *New Feature: brief description of feature*

## This PR adds/fixes...

<!-- List your features here and the benefits they bring. -->

## Notes

> <!-- List anything note-worthy here (potential issues, this needs to be merged to `master` before working, etc....). -->

Reviewer: ```suggestion
<!-- List anything note-worthy here (potential issues, this needs to be merged to `master` before working, etc.). -->
```

---

## Double check these details before you open a PR**

> *Tick the checkboxes to ensure you've done everything*

Reviewer: ```suggestion
*Tick the checkboxes to ensure you've done everything correctly*
```

- [] PR does not match another non-stale PR currently opened
- [] PR name matches the format *New Feature: brief description of feature*

## This PR adds/fixes...

> *List your features here and the benefits they bring.*

Reviewer: ```suggestion
<!-- List your features here and the benefits they bring. -->
```

Author: Why do we comment out this line though? Shouldn't the user see this?

Reviewer: > Why do we comment out this line, though? Shouldn't the user see this?

This works like in other programming languages: comments cannot be read by computers, but they can be read by the user. The purpose of the comments should only be read by the PR creator, not by other users. Hope I get to understand. 🙂

Author: I think that line should be there as an instruction to people opening a PR into the repo. If it's only readable to us, people might get confused. 

Will this line shows up when someone open a PR into the repo? If not, I think we should stick with the old one

Reviewer: Hmm, I think you misunderstood, sorry. 😅

When you use comments in Markdown, that's only readable when editing a comment or a Markdown file, but that can't be read when looking at the file/comment preview. Hope I get to understand now. 😅

Author: I see. In that case, I think we should stick with what we had. I want the people opening PRs to see that instruction and not have it hidden. Let me know if you had something else in mind for this. If not, I'll dismiss this in a few days

Reviewer: Hmm, I think you misunderstood again... 😅
Try to use this code in a comment:

```markdown
<!-- List your features here and the benefits they bring. -->
```

Before submitting it, click on the `Preview` button, check the content, and then go back to the `Write` button.

## Notes

> *List anything note-worthy here (potential issues, this needs to be merged to `master` before working, etc....).*
> *Don't forget to link any issues that this PR will solved.*
> 
> **Please link any issue that this PR should solve if it's applicable**

Reviewer: ```suggestion
<!-- List anything note-worthy here (potential issues, this needs to be merged to `master` before working, etc....).
Don't forget to link any issues that this PR will solve.

Please link any issue that this PR solves if applicable -->
```

Author: Same here for the comments. The verb tenses fixes are appreciated.
